core/state_processor_rip7560.go

```diff
@@ -211,8 +176,19 @@ func ApplyRip7560ValidationPhases(chainConfig *params.ChainConfig, bc ChainConte
 		GasPrice: gasPrice,
 	}
 	evm := vm.NewEVM(blockContext, txContext, statedb, chainConfig, cfg)
+	epc := &EntryPointCall{}
+
+	if evm.Config.Tracer == nil {
+		evm.Config.Tracer = &tracing.Hooks{
+			OnEnter: epc.OnEnter,
+		}
+	} else {
+		// keep the original tracer's OnEnter hook
+		epc.OnEnterSuper = evm.Config.Tracer.OnEnter
+		evm.Config.Tracer.OnEnter = epc.OnEnter
+	}
 
-	if evm.Config.Tracer != nil && evm.Config.Tracer.OnTxStart != nil {
+	if evm.Config.Tracer.OnTxStart != nil {
```

Comment:
unrelated to this PR: calling OnTxStart without OnTxEnd (and actually, it is not even clear if/when it should be called, as it depends on the meaning of "Tx" in this context: is it a frame, entire validation, or validation+execution)

Reply:
Yes, preferably we would end up modifying the tracer API to better suit the needs of RIP-7560. But that is beyond the current scope, I think.